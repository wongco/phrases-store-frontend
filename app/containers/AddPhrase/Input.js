import styled from 'styled-components';

const Input = styled.input`
  height: 2em;
  padding: 0 1em;
  width: 100%;
  border: 1px solid orange;
`;

export default Input;
