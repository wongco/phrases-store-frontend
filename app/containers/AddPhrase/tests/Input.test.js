import React from 'react';
import { enzymeFind } from 'styled-components/test-utils';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from '../Input';

describe('<Input />', () => {
  it('renders without crashing', () => {
    shallow(<Input />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<Input />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const disabled = true;
    const type = 'text';
    const wrapper = mount(<Input disabled type="text" />);
    const renderedComponent = enzymeFind(wrapper, Input);
    expect(renderedComponent.prop('disabled')).toBe(disabled);
    expect(renderedComponent.prop('type')).toBe(type);
  });
});
