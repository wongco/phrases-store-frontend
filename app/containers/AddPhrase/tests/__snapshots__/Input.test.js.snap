// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Input-sc-1o0je8j-0",
        "isStatic": false,
        "rules": Array [
          "height:2em;padding:0 1em;width:100%;border:1px solid ",
          [Function],
          ";",
          [Function],
          ";",
        ],
      },
      "displayName": "Input",
      "render": [Function],
      "styledComponentId": "Input-sc-1o0je8j-0",
      "target": "input",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
